kf_lib_data_ingest/etl/stage_analyses.py

```diff
@@ -57,28 +49,113 @@ def check_counts(discovery_sources, expected_counts):
                 elif key.UNIQUE_KEY in discovery_sources:
                     key = key.UNIQUE_KEY
 
-            found = uniques.get(key)
+            found = uniques_without_na.get(key)
             if expected == found:
-                passmark = '✅'
+                flag = '✅'
             else:
                 passed = False
-                passmark = '❌'
+                flag = '❌'
             checks = checks.append(
                 {
-                    'key': key, 'expected': expected, 'found': found,
-                    'equal': passmark
+                    'Key': key, 'Expected': expected, 'Found': found,
+                    'Equal': flag
                 },
                 ignore_index=True
             )
-        message.append(
+
+        if passed:
+            messages.append(f'✅ All counts are as expected.')
+        else:
+            messages.append(f'❌ Counts are not all as expected.')
+
+        messages.append(
+            'EXPECTED COUNT CHECKS:\n' +
             tabulate(
                 checks,
                 headers='keys', showindex=False, tablefmt='psql'
             )
         )
 
-    return passed, '\n'.join(message)
+    return passed, messages
 
 
-def compare_counts(discovery_sources_one, discovery_sources_two):
-    False, "Compare Counts NYI"
+def compare_counts(
+    name_one, discovery_sources_one, name_two, discovery_sources_two
+):
+    title = 'COMPARING COUNTS'
+    messages = [title + '\n' + '='*len(title)]
+    passed = True
+
+    one_keys = discovery_sources_one.keys()
+    two_keys = discovery_sources_two.keys()
+
+    comparison_df, diff_num = _compare(name_one, one_keys, name_two, two_keys)
+
+    if diff_num != 0:
+        msg = f'❌ Column names not equal between {name_one} and {name_two}'
+        messages.extend(_format(msg, comparison_df))
+        passed = False
+    else:
+        msg = f'✅ Column names are equal between {name_one} and {name_two}'
+        messages.append(msg)
+        messages.append(SEP)
+
+    good_msg_ag = []
+    bad_msg_ag = []
+    for k in (one_keys | two_keys):
```

Comment:
Should this be 
```suggestion
    for k in set(one_keys | two_keys):
```

Reply:
not necessary. set operations work on dict_keys